CHANGELOG.md

```diff
@@ -1,5 +1,36 @@
 # go-ipfs changelog
 
+### 0.3.4 - unreleased
+
+#### API changes
+
+##### `core/coreunix`'s `Add` → `shell/fsnode`'s `AddFromReader`
+
+We've added an `AddFromReader` function to `shell/fsutils`.  The old
+`Add` from `core/coreunix` is deprecated and will be removed in
+version 0.4.0. To update your existing code, change usage like:
+
+    keyString, err := coreunix.Add(ipfsNode, reader)
+    if err != nil {
+        return err
+    }
+
+to
+
+    fileNode, err := fsutils.AddFromReader(ipfsNode, reader)
+    if err != nil {
+        return err
+    }
+    key, err := fileNode.Key()
+    if err != nil {
+        return err
+    }
+    keyString := key.String()
+
+That's a bit more verbose if all you want is the stringified key, but
+returning a `dag.Node` makes it easier to perform other operations
+like adding the file node to a directory node.
```

Comment:
On Wed, Apr 29, 2015 at 01:26:36PM -0700, Juan Batiz-Benet wrote:
> the binary's changelog is not the best place. this changelog is just
> a tiny summary. maybe we can have a different file for things like
> this.

Git uses a directory of per-release notes [1].  I'm happy to put this
in docs/release-notes/unreleased.md (to be moved to
docs/release-notes/0.3.4.md when you cut that release) if that sounds
reasonable to you.

[1]: http://git.kernel.org/cgit/git/git.git/tree/Documentation/RelNotes?id=v2.3.7